packages/screens/Rakki/components/BuyTicketsButton.tsx

```diff
@@ -0,0 +1,375 @@
+import { BrandText } from "@/components/BrandText";
+import { OptimizedImage } from "@/components/OptimizedImage";
+import { SVG } from "@/components/SVG";
+import { Box } from "@/components/boxes/Box";
+import { PrimaryButton } from "@/components/buttons/PrimaryButton";
+import { SecondaryButton } from "@/components/buttons/SecondaryButton";
+import { MainConnectWalletButton } from "@/components/connectWallet/MainConnectWalletButton";
+import { GradientText } from "@/components/gradientText";
+import { DepositWithdrawModal } from "@/components/modals/DepositWithdrawModal";
+import ModalBase from "@/components/modals/ModalBase";
+import { useFeedbacks } from "@/context/FeedbacksProvider";
+import { ExecMsg, Info } from "@/contracts-clients/rakki/Rakki.types";
+import { useBalances } from "@/hooks/useBalances";
+import useSelectedWallet from "@/hooks/useSelectedWallet";
+import {
+  NetworkFeature,
+  getNetworkFeature,
+  getCosmosNetwork,
+  CurrencyInfo,
+} from "@/networks";
+import { getKeplrSigningCosmWasmClient } from "@/networks/signer";
+import { prettyPrice } from "@/utils/coins";
+import {
+  errorColor,
+  neutral00,
+  neutral17,
+  neutral22,
+  neutral33,
+  neutral77,
+  neutralA3,
+  neutralFF,
+} from "@/utils/style/colors";
+import {
+  fontSemibold13,
+  fontSemibold14,
+  fontSemibold16,
+} from "@/utils/style/fonts";
+import { layout } from "@/utils/style/layout";
+import { modalMarginPadding } from "@/utils/style/modals";
+
+export const BuyTicketsButton: FC<{ networkId: string; info: Info }> = ({
+  networkId,
+  info,
+}) => {
+  const selectedWallet = useSelectedWallet();
+  const [modalVisible, setModalVisible] = useState(false);
+  const remainingTickets = info.config.max_tickets - info.current_tickets_count;
+  const [ticketAmount, setTicketAmount] = useState("1");
+  const queryClient = useQueryClient();
+  const ticketAmountNumber = Long.fromString(ticketAmount || "0");
+  useEffect(() => {
+    if (remainingTickets > 0 && ticketAmountNumber.gt(remainingTickets)) {
+      setTicketAmount(remainingTickets.toString());
+    }
+  }, [ticketAmountNumber, remainingTickets]);
+  const totalPrice = ticketAmountNumber.mul(
+    Long.fromString(info.config.ticket_price.amount),
+  );
+  const { balances } = useBalances(networkId, selectedWallet?.address);
+  const ticketDenomBalance =
+    balances.find((b) => b.denom === info.config.ticket_price.denom)?.amount ||
+    "0";
+  const canPay = Long.fromString(ticketDenomBalance).gte(totalPrice);
+  const canBuy = ticketAmountNumber.gt(0) && canPay;
+  const { wrapWithFeedback } = useFeedbacks();
+
+  const [isDepositVisible, setDepositVisible] = useState(false);
+  const network = getCosmosNetwork(networkId);
+  const atomIbcCurrency = useMemo(() => {
+    return network?.currencies.find(
+      (currencyInfo: CurrencyInfo) =>
+        currencyInfo.kind === "ibc" && currencyInfo.sourceDenom === "uatom",
+    );
+  }, [network]);
```

Comment:
why hardcode atom? we should get the currency from the rakki config

Reply:
The contract's config provides utori denom.
The Deposit action is a way for the user to get TORI. By default, we use ATOM as source.
I used the same behavior than on TopMenu: https://github.com/TERITORI/teritori-dapp/blob/30862013820222bf2ce193263ec2c8ba0c93d5f5/packages/components/TopMenu/TopMenuMyWallets.tsx#L196-L201

Should we add a "deposit_denom" in the contract ?

Reply:
I removed that. It's out of scope. We could merge this PR soon